src/laser_scan_assembler.cpp

```diff
@@ -117,16 +314,171 @@ class LaserScanAssembler : public BaseAssembler<sensor_msgs::LaserScan>
     }
   }
 
+  bool startCollection(StartCollection::Request& req, StartCollection::Response& resp)
+  {
+    scan_hist_mutex_.lock();
+    scan_hist_.clear();
+    scan_hist_mutex_.unlock();
+
+    current_req_ = req;
+    ROS_WARN_COND(req.horizontal_resolution == 0, "horizontal_resolution is zero");
+    ROS_WARN_COND(req.vertical_resolution == 0, "vertical_resolution is zero");
+
+    scan_range_buffer_ = cv::Mat::zeros(1000, 1000, CV_16UC1);
```

Comment:
If we initialize at the first scan, there is no reason to oversize it first and then crop it later